swarm_event_bus_test.go

```diff
@@ -0,0 +1,170 @@
+package swarm_test
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peerstore"
+
+	"github.com/libp2p/go-eventbus"
+	. "github.com/libp2p/go-libp2p-swarm"
+	swarmt "github.com/libp2p/go-libp2p-swarm/testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNotifieeEventbusSimple(t *testing.T) {
+	ctx := context.Background()
+	s1 := swarmt.GenSwarm(t, ctx)
+	defer s1.Close()
+	s2 := swarmt.GenSwarm(t, ctx)
+	defer s2.Close()
+
+	// subscribe for notifications on s1
+	s, err := s1.EventBus().Subscribe(&event.EvtPeerConnectednessChanged{})
```

Comment:
tests in the package can directly access `s1.bus`, and don't need the additional interface method.

Reply:
@willscott Aha ! Except, the swarm tests are in a different package 🍡